storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -2226,12 +2233,12 @@ void get_id_from_mysql_table_scan(std::string name, uint64 *id, THD *thd) {
       break;
     }
 
+    mtr_start(&mtr);
     rec = (rec_t *)dd_getnext_system_rec(&pcur, &mtr);
   }
 
-  ut_ad(*id != 0);
+  ut_a(*id != 0);
 
-  mtr_commit(&mtr);
```

Comment:
isn't this mtr_commit required?

Reply:
original dd processing function does mtr commit at end

Reply:
in our case, It should always break. so that's why I have removed mtr_commit before dd_table_close. It would be same thing adding mtr_start here and followed by mtr_commit before dd_table_close. 
Since we have assert before dd_table_close(in case it doesn't break)

Reply:
if it doesn't break, we need the mtr_commit() ?